File: recommender.cpp
#include "recommender.hpp"

Recommender::Recommender(long long UsersSize, long long ItemsSize, long long Dimension)
	: Users(cv::Mat(UsersSize, Dimension, CV_R))
	, Items(cv::Mat(Dimension, ItemsSize, CV_R))
	, Dimension(Dimension)
{
}

R Recommender::Predict(UID uid, IID iid)
{
	return Users.row(uid).dot(Items.col(iid).t());
}

File: python/recommender.py
import numpy as np

class Recommender(object):
    def __init__(self, UsersSize, ItemSize, Dimension):
        self.users = np.zeros((UsersSize, Dimension))
        self.items = np.zeros((Dimension, ItemSize))
        self.d = Dimension

    def predict(self, uid, iid):
        return np.dot(self.users[uid, :], self.items[:, iid])
